### PR TITLE
Add recipe to change the image name, including repo and tag

### PR DESCRIPTION
**Pull request by jbrisbin**

### CODE REVIEW

File: src/main/java/org/openrewrite/kubernetes/opa/UpgradeContainerImage.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+                                int slashIdx = currentImageName.indexOf('/');
+                                String currentRepoName = slashIdx > -1
+                                        ? currentImageName.substring(0, slashIdx)
+                                        : "";
+                                boolean hasAllowedRepo = Arrays.binarySearch(allowedRepos, currentRepoName) > -1;
+
+                                int colonIdx = currentImageName.lastIndexOf(':');
+                                String currentVersionTag = colonIdx > -1
+                                        ? currentImageName.substring(colonIdx + 1)
```
Comment (jkschneider) [inline src/main/java/org/openrewrite/kubernetes/opa/UpgradeContainerImage.java:98]: our code style is to put operators on the same line not next.

File: src/test/kotlin/org/openrewrite/kubernetes/opa/UpgradeContainerImageTest.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            kind: Pod
+            metadata:
+              labels:
+                app: application
+            spec:
+              containers:            
+              - name: mycontainer
+                image: repo.dev.lan/nginx:1.2.3
+        """,
+        cycles = 2
```
Comment (jkschneider) [inline src/test/kotlin/org/openrewrite/kubernetes/opa/UpgradeContainerImageTest.kt:53]: I believe cycles=2 is the default?

File: src/main/java/org/openrewrite/kubernetes/opa/UpgradeContainerImage.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                                }
+
+                                int slashIdx = currentImageName.indexOf('/');
+                                String currentRepoName = slashIdx > -1 ? currentImageName.substring(0, slashIdx) : "";
+                                boolean hasAllowedRepo = allowedRepos.contains(currentRepoName);
+
+                                int colonIdx = currentImageName.lastIndexOf(':');
+                                String currentVersionTag = colonIdx > -1 ? currentImageName.substring(colonIdx + 1) : "";
+
+                                String newImageName = (currentRepoName.isEmpty() || !hasAllowedRepo ? preferredRepo : currentRepoName)
```
Comment (jbrisbin) [inline src/main/java/org/openrewrite/kubernetes/opa/UpgradeContainerImage.java:95]: should this also support setting the repo back to "" in the case of removing this change?

**Comment by jbrisbin** [comment]
Going to take a slight different approach.

**Pull request closed without merging**